darts/models/torch_forecasting_model.py

```diff
@@ -582,142 +614,53 @@ def predict_from_dataset(self,
         Sequence[TimeSeries]
             Returns one or more forecasts for time series.
         """
-        self.model.eval()
+        self._verify_inference_dataset_type(input_series_dataset)
 
         if roll_size is None:
             roll_size = self.output_chunk_length
         else:
             raise_if_not(0 < roll_size <= self.output_chunk_length,
                          '`roll_size` must be an integer between 1 and `self.output_chunk_length`.')
 
-        # check input data type
-        raise_if_not(isinstance(input_series_dataset, TimeSeriesInferenceDataset),
-                     'Only TimeSeriesInferenceDataset is accepted as input type.')
-
         # check that `num_samples` is a positive integer
         raise_if_not(num_samples > 0, '`num_samples` must be a positive integer.')
 
         # iterate through batches to produce predictions
         batch_size = batch_size or self.batch_size
-        pred_loader = DataLoader(TimeSeriesTorchDataset(input_series_dataset, self.device),
+
+        pred_loader = DataLoader(input_series_dataset,
                                  batch_size=batch_size,
                                  shuffle=False,
                                  num_workers=0,
                                  pin_memory=False,
                                  drop_last=False)
         predictions = []
         iterator = _build_tqdm_iterator(pred_loader, verbose=verbose)
+
+        self.model.eval()
         with torch.no_grad():
             for batch_tuple in iterator:
 
-                # at this point `input_series` contains both the past target series and past covariates
-                input_series = batch_tuple[0].to(self.device)
-                cov_future = batch_tuple[1] if len(batch_tuple) == 3 else None
+                input_data_tuple, batch_input_series = batch_tuple[:-1], batch_tuple[-1]
```

Comment:
Here I think we need to move those batches with `.to(self.device)`. Otherwise, we could delegate this responsibility to `self._get_batch_prediction`, but in this last case we would have to remind to move tensors in every implementation of `self._get_batch_prediction` when we are adding a new model.  
```suggestion
                input_data_tuple, batch_input_series = batch_tuple[:-1].to(self.device), batch_tuple[-1].to(self.device)
```

Reply:
This is now done in the function `TorchForecastingModel._batch_collate_fn()`. For all Numpy arrays returned by the datasets, we stack them into batches, translate to torch Tensors, and finally put on the right device. This function is applied both to training and inference datasets, so I think it's fine like that.